webpack/scenes/ContentViews/Delete/Steps/CVDeletionReassignActivationKeysForm.js

```diff
@@ -0,0 +1,121 @@
+import useDeepCompareEffect from 'use-deep-compare-effect';
+import { ExpandableSection, Select, SelectOption } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { STATUS } from 'foremanReact/constants';
+import getContentViews from '../../ContentViewsActions';
+import { selectContentViewError, selectContentViews, selectContentViewStatus } from '../../ContentViewSelectors';
+import CVDeleteContext from '../CVDeleteContext';
+import EnvironmentPaths from '../../components/EnvironmentPaths/EnvironmentPaths';
+import AffectedActivationKeys from '../../Details/Versions/Delete/affectedActivationKeys';
+
+const CVDeletionReassignActivationKeysForm = () => {
+  const dispatch = useDispatch();
+  const contentViewsInEnvResponse = useSelector(state => selectContentViews(state, 'keys'));
+  const contentViewsInEnvStatus = useSelector(state => selectContentViewStatus(state, 'keys'));
+  const contentViewsInEnvError = useSelector(state => selectContentViewError(state, 'keys'));
+  const cvInEnvLoading = contentViewsInEnvStatus === STATUS.PENDING;
+  const [cvSelectOpen, setCVSelectOpen] = useState(false);
+  const [cvSelectOptions, setCvSelectionOptions] = useState([]);
+  const [showActivationKeys, setShowActivationKeys] = useState(false);
+  const {
+    currentStep, selectedEnvForAK, cvEnvironments, cvId,
+    setSelectedEnvForAK, selectedCVForAK, setSelectedCVNameForAK, setSelectedCVForAK,
+  } = useContext(CVDeleteContext);
+
+  useDeepCompareEffect(
+    () => {
+      if (selectedEnvForAK.length) {
+        dispatch(getContentViews({
+          environment_id: selectedEnvForAK[0].id,
+          include_default: true,
+          full_result: true,
+        }, 'keys'));
+      }
+      setCVSelectOpen(false);
+    },
+    [selectedEnvForAK, dispatch, setCVSelectOpen],
+  );
+
+  useDeepCompareEffect(() => {
+    const { results } = contentViewsInEnvResponse;
+    const contentViewEligible = cv => cv.id !== cvId;
+    if (!cvInEnvLoading && results && selectedEnvForAK.length) {
+      setCvSelectionOptions(results.map(cv => ((contentViewEligible(cv)) ?
+        (
+          <SelectOption
+            key={cv.id}
+            value={cv.id}
+          >
+            {cv.name}
+          </SelectOption>
+        ) : null)).filter(n => n));
+    }
+    if (!cvInEnvLoading && results && selectedCVForAK &&
+      results.filter(cv => cv.id === selectedCVForAK && contentViewEligible(cv)).length === 0) {
+      setSelectedCVForAK(null);
+      setSelectedCVNameForAK(null);
+    }
+  }, [contentViewsInEnvResponse, contentViewsInEnvStatus, currentStep,
+    contentViewsInEnvError, selectedEnvForAK, setSelectedCVForAK, setSelectedCVNameForAK,
+    cvInEnvLoading, selectedCVForAK, cvId]);
+
+  const fetchSelectedCVName = (id) => {
+    const { results } = contentViewsInEnvResponse ?? { };
+    return results.filter(cv => cv.id === id)[0]?.name;
+  };
+
+  const onSelect = (event, selection) => {
```

Comment:
_event

Reply:
Wonder why lint doesn't complain about this..